Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,293 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+    using Microsoft.ApplicationInsights.Channel;
+
+    internal class TelemetryDiagnosticSourceListener : IObserver<DiagnosticListener>, IDisposable
+    {
+        private const string ActivityNameSuffix = ".Monitoring";
+        private const string StopSuffix = ".Stop";
+        private const string ActivityStopNameSuffix = ActivityNameSuffix + StopSuffix;
+
+        private readonly TelemetryClient client;
+        private readonly TelemetryConfiguration configuration;
+
+        private readonly IDisposable listenerSubscription;
+        private List<IDisposable> individualSubscriptions;
+
+        public TelemetryDiagnosticSourceListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceListener + ":");
+
+            this.configuration = configuration;
+
+            try
+            {
+                this.listenerSubscription = DiagnosticListener.AllListeners.Subscribe(this);
+            }
+            catch (Exception ex)
+            {
+                DependencyCollectorEventSource.Log.TelemetryDiagnosticSourceListenerFailedToSubscribe(ex.ToInvariantString());
+            }
+        }
+
+        /// <summary>
+        /// This method gets called once for each existing DiagnosticListener when this
+        /// DiagnosticListener is added to the list of DiagnosticListeners
+        /// (<see cref="System.Diagnostics.DiagnosticListener.AllListeners"/>). This method will
+        /// also be called for each subsequent DiagnosticListener that is added to the list of
+        /// DiagnosticListeners.
+        /// <seealso cref="IObserver{T}.OnNext(T)"/>
+        /// </summary>
+        /// <param name="value">The DiagnosticListener that exists when this listener was added to
+        /// the list, or a DiagnosticListener that got added after this listener was added.</param>
+        public void OnNext(DiagnosticListener value)
+        {
+            if (value != null)
+            {
+                IDisposable subscription = value.Subscribe(
+                    new IndividualDiagnosticSourceListener(value, this),
+                    (evnt, r, _) => evnt.EndsWith(ActivityNameSuffix, StringComparison.OrdinalIgnoreCase)
+                        || evnt.EndsWith(ActivityStopNameSuffix, StringComparison.OrdinalIgnoreCase));
```

Comment:
we also may receive an exception from the same source.  Should be useful to report them.

Reply:
makes sense, although that would be Exception telemetry collection and for now I'd like to limit this PR to Dependency collection

Reply:
I'm fine with implementing it later. 
Though the same listener should handle Exception events as well, it cannot be done by other listeners (or would be quite inefficient)

Reply:
Agreed :)